oauthenticator/oauth2.py

```diff
@@ -291,9 +293,9 @@ class OAuthenticator(Authenticator):
 
         .. warning::
 
-           When this is enabled and you are to remove access for one or more
-           users allowed via other config options, you must make sure that they
-           are not part of the database of users still. This can be tricky to do
+           When this is enabled and you wish to remove access for one or more
+           users previously allowed, you must make sure that they
+           are not removed from the jupyterhub database. This can be tricky to do
```

Comment:
```suggestion
           are removed from the jupyterhub database. This can be tricky to do
```

Reply:
Lower down in this help message there's a note
```
        .. note::
           Allowing existing users is done by adding existing users on startup
           and newly created users to the `allowed_users` set. Due to that, you
           can't rely on this config to independently allow existing users if
           you for example would reset `allowed_users` after startup.

        .. versionadded:: 16.0

        .. versionchanged:: 16.0
```
After reading it several times I still don't understand what the second sentence means. Do you have any thoughts on how to make it clearer?

Reply:
Its probably me trying to convey the hidden coupling of allow_existing_users and allowed_users, which could come as a surprise if looking at allowed_users after startup.

The mechanism for allow_existing_users to allow existing users is through appending users to allowed_users. I wanted this complexity surfaced to be an answer to someone unexpectedly running into confusion about this in some way - maybe via functionality in https://github.com/jupyterhub/jupyterhub/pull/4442